File: 2022/day4/star2.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <sstream>
int main() 
{
    std::ifstream input("input.txt");
    std::string line;
    int sum = 0, a, b, x, y;
    while(getline(input, line)) 
    {
        std::stringstream ss(line);
        char c;
        ss >> a >> c >> b >> c >> x >> c >> y;

        sum+=(a<=x && b>=x) || (x<=a && y>=a);

    }
    std::cout << sum << std::endl;
}

File: 2025/day6/star2.py
from math import prod
ans = 0

nums = []

with open("2025/day6/input.txt") as f:
    for line in f:
        if line[0].isnumeric():
            nums.append(line.strip())
        else:
            operations=line.strip().split()

terms = [[] for _ in range(len(operations))]
op_num = 0
for i in range(len(nums[0])):
    cur = -1
    for j in range(len(nums)):
        if nums[j][i].isnumeric():
            cur = (cur*10 + int(nums[j][i])) if cur!=-1 else int(nums[j][i])
    if cur == -1:
        op_num+=1
    else:
        terms[op_num].append(cur)

for i in range(len(operations)):
    ans = ans + (sum(terms[i]) if operations[i]=='+' else prod(terms[i]))

print(ans)
    
